lib/AST/ASTImporter.cpp

```diff
@@ -4772,14 +4797,14 @@ Decl *ASTNodeImporter::VisitClassTemplateSpecializationDecl(
         return nullptr;
       CanonInjType = CanonInjType.getCanonicalType();
 
-      TemplateParameterList *ToTPList = ImportTemplateParameterList(
-            PartialSpec->getTemplateParameters());
-      if (!ToTPList && PartialSpec->getTemplateParameters())
+      auto ToTPListOrErr = ImportTemplateParameterList(
```

Comment:
Perhaps instead of `auto` it is better to explicitly say that this is an `Error`. (Or is it an `Expected` ? :) )
I'd use `auto` only if the same statement contains the explicit type like in case of `cast<>`.

Reply:
I wanted to use `Error` if `Error` is returned, but `auto` if an `Expected<something-that-maybe-long>` is returned to make the code simpler.

Reply:
Ok.